app/scripts/components/common/catalog/catalog-content.tsx

```diff
@@ -0,0 +1,339 @@
+import React, { useState, useMemo, useEffect, useCallback } from 'react';
+import styled from 'styled-components';
+import { DatasetData } from 'veda';
+import { useNavigate } from 'react-router-dom';
+import { glsp, themeVal } from '@devseed-ui/theme-provider';
+import TextHighlight from '../text-highlight';
+import { CollecticonDatasetLayers } from '../icons/dataset-layers';
+import prepareDatasets from './prepare-datasets';
+import FiltersControl from './filters-control';
+import { CatalogCard } from './catalog-card';
+import CatalogTagsContainer from './catalog-tags';
+
+import { CatalogActions } from './utils';
+import { CardList } from '$components/common/card/styles';
+import EmptyHub from '$components/common/empty-hub';
+import { DATASETS_PATH } from '$utils/routes';
+import {
+  getTaxonomyByIds,
+  generateTaxonomies,
+  getTaxonomy,
+  TAXONOMY_SOURCE,
+} from '$utils/veda-data';
+import { OptionItem } from '$components/common/form/checkable-filter';
+import { findParentDataset, getAllDatasetsWithEnhancedLayers } from '$components/exploration/data-utils';
+import { Pill } from '$styles/pill';
+import { usePreviousValue } from '$utils/use-effect-previous';
+
+const EXCLUSIVE_SOURCE_WARNING = "Can only be analyzed with layers from the same source";
+
+export interface CatalogContentProps {
+  datasets: DatasetData[];
+  selectedIds?: string[];
+  setSelectedIds?: (selectedIds: string[]) => void;
+  filterLayers?: boolean;
+  emptyStateContent?: React.ReactNode;
+  search: string;
+  taxonomies: Record<string, string[]>;
+  onAction: (action: CatalogActions, value?: any) => void;
+}
+
+const DEFAULT_SORT_OPTION = 'asc';
+
+function CatalogContent({
+  datasets,
+  selectedIds,
+  setSelectedIds,
+  filterLayers,
+  emptyStateContent,
+  search,
+  taxonomies,
+  onAction,
+}: CatalogContentProps) {
+  const [exclusiveSourceSelected, setExclusiveSourceSelected] = useState<string | null>(null);
+  const isSelectable = selectedIds !== undefined;
```

Comment:
It might be better to separate selectable card component  from this component since the behavior is pretty different, and the conditional blocks are pretty big. But not a blocker for now.

Reply:
Yeah I started it like that, then merged them into one. Should've probably kept it separate.